### PR TITLE
Login react

### PR DESCRIPTION
**Pull request by nirjacob**
* Added Login page

### CODE REVIEW

File: controllers/login.js [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
-        }
-    } catch (err) {
-        return res.status(400).send(`Login failed, ${err.message}`)
+  try {
+    const { username, password } = req.body
+    if (await authenticateLogin(username, password)) {
+      const newToken = createNewJwt(username)
+      return res.status(200).send(newToken)
+    } else {
+      return res.status(400).send('Login failed, username or password incorrect')
```
Comment (racheli-lavan) [inline controllers/login.js:11]: why 400? 400 is bad request, this is not the case here. just do:
` return next(new Error('Login failed, username or password incorrect'))`

File: controllers/login.js [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
     }
+  } catch (err) {
+    return res.status(400).send(`Login failed, ${err.message}`)
+  }
```
Comment (racheli-lavan) [inline controllers/login.js:15]: here also - return `next(err)`

File: react-client/src/containers/Login/Login.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import { submitLogin } from '../../services/Podcasts'
+import { useNavigate } from 'react-router-dom'
+
+const Login = () => {
+  const [loginDetails, setLoginDetails] = useState('')
+  const [loginStatus, setLoginStatus] = useState('')
+  const navigate = useNavigate()
+
+  useEffect(() => {
+    document.body.style.background = 'linear-gradient(to right, #11998e 0%, #38ef7d 100%)'
```
Comment (racheli-lavan) [inline react-client/src/containers/Login/Login.js:15]: Please put styling in css and not here

File: react-client/src/containers/Login/Login.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        ...prevValue,
+        [name]: value
+      }
+    })
+  }
+
+  async function submitLoginDetails() {
+    try {
+      window.authorization = await submitLogin(loginDetails)
+      setLoginStatus('')
```
Comment (racheli-lavan) [inline react-client/src/containers/Login/Login.js:36]: it's not the best practice to put a ui message as state data and call it status. it doesn't represent a status. it represent a message.
I would add to state a boolean called isLoginError  with a default of false. when set to true- the ui will show the error message of "Incorrect username or password" (an element that this is it's text)

Comment (nirjacob) [reply react-client/src/containers/Login/Login.js:36]: Using simple boolean doesn't re-render the element so I used a useState boolean

**Pull request merged**